src/nvim/charset.c

```diff
@@ -1655,88 +1658,66 @@ void vim_str2nr(const char_u *const start, int *const prep, int *const len,
 
       if (what & STR2NR_OCT) {
         // Don't interpret "0", "08" or "0129" as octal.
-        for (int n = 1; ascii_isdigit(ptr[n]); ++n) {
-          if (ptr[n] > '7') {
+        for (int i = 1; !STRING_ENDED(ptr + i) && ascii_isdigit(ptr[i]); i++) {
+          if (ptr[i] > '7') {
             // can't be octal
             pre = 0;
             break;
           }
-          if (ptr[n] >= '0') {
+          if (ptr[i] >= '0') {
```

Comment:
This condition is always true.

Reply:
Wondering why it was there in the first place then and what was supposed to mean.

Vim has it as well though, so can ask there.